code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -52,11 +54,13 @@
 	return ..()
 
 /obj/effect/particle_effect/smoke/proc/kill_smoke()
-	if(CHECK_BITFIELD(smoke_traits, SMOKE_CAMO))
-		apply_smoke_effect(get_turf(src))
 	LAZYCLEARLIST(cloud?.smoked_mobs)
 	STOP_PROCESSING(SSobj, src)
 	INVOKE_ASYNC(src, .proc/fade_out)
+	if(CHECK_BITFIELD(smoke_traits, SMOKE_CAMO))
+		var/turf/T = get_turf(src)
+		if(T)
```

Comment:
Alright, the idea here is to trace back to the origin of the problem, instead of silencing it later on.
In here (and in the other times) `get_turf(src)` would return `null` if this was in nullspace. Such as when it's being QDELETED. Can you think of another alternative?
If it's being QDELETED, why is that happening? That's the question.

Reply:
I can guess the problem with the smoke being in nullspace to start with may be the holder happening to be in null space too. The safety check on start() is limited to the affixed location or holder being qdeleted or not as of now.
I'll open up another PR to add safety check and warning on the smoke spread datum start(), will need a TM to find the real origin of this issue.

Reply:
#1596. Keeping this PR open the time being should the issue not relate to the datum's start()